### PR TITLE
Add support for Durable Objects

### LINKED ISSUE #7: Support Durable Objects

### PR DESCRIPTION
**Pull request by jamesopstad**
resolves #7

This PR adds support for Durable Objects. They can be defined in the same worker or another worker (using `script_name`). RPC is also supported.

### CODE REVIEW

File: packages/vite-plugin-cloudflare/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+				const classNames = durableObjectClassNames[worker.name];
+				invariant(classNames, errorMessage(worker.name));
+
+				classNames.add(value);
+			} else if (typeof value === 'object') {
+				if (value.scriptName) {
+					const classNames = durableObjectClassNames[value.scriptName];
+					invariant(classNames, errorMessage(value.scriptName));
+
+					classNames.add(value.className);
```
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/utils.ts:62]: here, are you taking the `scriptName` and using it as the `className`?

this doesn't seem correct to me... is it? 😮

Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/utils.ts:62]: Here `scriptName` is the name of the worker where the DO is defined. So we get a reference to the `Set` of exports for that worker and then add the `className` to it.

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ interface WebSocketContainer {
 	webSocket?: WebSocket;
 }
 
+const webSocketError = 'The WebSocket is undefined';
```
[the remark this answers is not in the document]
Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:16]: Fine with me.

File: packages/vite-plugin-cloudflare/src/utils.ts (hunk 1 above)
Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/utils.ts:62]: Maybe it's the naming that's confusing. I could call the variables `entryWorkerEntrypointNames` and `entryDurableObjectClassNames` to make it clearer that they are maps of entry files to exports.

Comment (dario-piotrowicz) [reply packages/vite-plugin-cloudflare/src/utils.ts:62]: yeah by debugging it I can see the logic there... but without context the variable names could be a bit confusing 🤔 

I personally would rename the variable and function and all that's related from a generic `DurableObjectClassNames` to something like `WorkersToDurableObjectClassNamesMap` or something like that, it is much more verbose but I personally feel like it makes things much much clearer here 🤔

**Comment by jamesopstad** [comment]
Thanks Dario! I've made all the changes you suggested.

**Pull request merged**